pkg/controller/multi_writer_at.go

```diff
@@ -35,14 +36,16 @@ func (m *MultiWriterError) Error() string {
 
 func (m *MultiWriterAt) WriteAt(p []byte, off int64) (int, error) {
 	errs := make([]error, len(m.writers))
+	wbs := make([]int, len(m.writers))
 	wg := sync.WaitGroup{}
 
 	for i, w := range m.writers {
 		wg.Add(1)
 		go func(index int, w io.WriterAt) {
-			_, err := w.WriteAt(p, off)
+			wn, err := w.WriteAt(p, off)
```

Comment:
IIRC, the `diffDisk` will align the write bytes. For example, application writes the range from 4 bytes to 514 bytes, `diffDisk` will write from 0 to 1023 bytes. Will we get correct `wn` here, i.e., 514 bytes? Or can we only get 1024 bytes? If it is the latter case, we need to do modification to make sure the returned value is correct.

Reply:
Could you help clarify the questions and update your findings before resolving the comment?

Reply:
Perhaps that I only use `dd` command to write data, it always writes 4096 bytes as the smallest size, and the offset is 0. I don't get into the `align the write bytes` scenario.
But we need to prevent data inconsistency if this scenario occurs.
Therefore, `WriteAt` and `readModifyWrite` are modified to get the correct write bytes.

Reply:
I did not find a good way to test the aligned IO.
Therefore, after discussing, we decided to have a generic way to collect the written bytes.
1. Get written bytes when encountering an error in the diffDisk layer instead of returning 0, for direct/align IO (computing read buf written with an offset).
2. Add a field in MutilWriterError/replicator to collect written bytes for each replica that encounters an error.
3. Retain proper RW replicas in the controller when replicas encounter the ENOSPC error.